### ISSUE TITLE
Add metrics around remote_read queries

### ISSUE THREAD

**Issue by amitabh91** [prometheus/prometheus#7242]
I have integrated M3DB as remote storage for Prometheus as a long term storage solution. Where M3coordinator runs as a side along with Prometheus which pushes the data to the M3DB cluster. I am able to write the metrics data into M3DB successfully. 

On executing the query prometheus_remote_storage_sent_bytes_total from Prom UI we see some incremental values, unlike prometheus_remote_storage_remote_read_queries. 

I am attaching the screenshot below for the quick reference 

**Write Data Logs**
![image](https://user-images.githubusercontent.com/7939820/81582108-41382a00-93cd-11ea-8a5d-eb67582e3894.png)

**Read Data Logs**
![image](https://user-images.githubusercontent.com/7939820/81582208-6167e900-93cd-11ea-8e68-d62b13956f32.png)

**Environment**
* System information:

   Linux 3.10.0-957.el7.x86_64 x86_64

* Prometheus version:

  prometheus, version 2.15.2 
  build user:       root@688433cf4ff7
  build date:       20200106-14:50:51
  go version:       go1.13.5

* Prometheus configuration file:
```
remote_write:
- url: http://localhost:7201/api/v1/prom/remote/write
  remote_timeout: 30s
  queue_config:
    capacity: 500
    max_shards: 1000
    min_shards: 1
    max_samples_per_send: 100
    batch_send_deadline: 5s
    min_backoff: 30ms
    max_backoff: 100ms
remote_read:
- url: http://localhost:7201/api/v1/prom/remote/read
  remote_timeout: 1m
  read_recent: true
```
Whenever we have mismatch in the  namespace value of  m3coordinator and M3DB cluster we see the below error while reading.

**m3coordinaor config**
```
clusters:
  - namespaces:
      - namespace: default
        retention: 48h
        type: unaggregated
    client:
      config:
        service:
          env: default_env
          zone: embedded
          service: m3db
          cacheDir: /var/lib/m3kv
          etcdClusters:
            - zone: embedded
              endpoints:
                - http://etcd.infra.svc.cluster.local:2379
```
**Error** 
![image](https://user-images.githubusercontent.com/7939820/81577182-c409b680-93c6-11ea-89e2-05cbe9bbec2f.png)

And with the correct namespace we don't see the error but prometheus_remote_storage_remote_read_queries still remains empty. So I believe this is a bug that needs a fix.

**Note**:- I have added m3query as datasource in Grafana and able to read the data stored in M3DB cluster.

**Comment by roidelapluie** [comment]
Thanks for the report.

prometheus_remote_storage_remote_read_queries is the number of remote read queries "in flight". Not the total of queries.

**Comment by amitabh91** [comment]
@roidelapluie  Thanks for responding. I have started working on Prometheus 2 weeks ago. So Kindly correct me if I am wrong.

"_prometheus_remote_storage_remote_read_queries is the number of remote read queries "in flight"_ "--- Could you please elaborate? Does it mean it will remain empty even when we are reading the data from remote storage? If so please let me know what query I should hit in Prom UI to get the remote read data.

Secondly, I mean  prometheus_remote_storage_succeeded_samples_total also shows positive values.

![image](https://user-images.githubusercontent.com/7939820/81742488-d0bd0600-94bd-11ea-8190-3fac9f2317df.png)

**Comment by roidelapluie** [comment]
It will be 1 if one remote read query is *ongoing*. Once the queries are finished, it will go back to 0.

**Comment by cstyan** [comment]
As Julien mentioned the metric shows the # of currently executing remote read queries, not the total number of remote read queries ever executed.

`prometheus_remote_storage_succeeded_samples_total` is not connected to remote read. That metric is used in remote write to track how many samples have successfully been sent. The label in your screenshot indicates that was well `api/v1/prom/remote/write`.

**Comment by amitabh91** [comment]
@roidelapluie Thanks for the clarification.   

@cstyan I understand prometheus_remote_storage_succeeded_samples_total is related to remote write not with remote read.
Its all about getting the counter value incremented for remote write but not for the remote read.

**Comment by amitabh91** [comment]
@roidelapluie  
I checked m3db pod logs and I can the query request. So ideally at the same timestamp we should see  prometheus_remote_storage_remote_read_queries=1. Attahcing logs and Prom UI querry screenshots below for your reference
```
{"level":"info","ts":1589354673.473276,"msg":"finished handling request","rqID":"904daafc-8d98-419a-9ce4-6520e96cd388","time":1589354673.4732697,"response":1.706975705,"url":"/api/v1/series?match[]=container_cpu_usage_seconds_total&start=1589311474&end=1589354674"}

```
**Prom UI** 

![image](https://user-images.githubusercontent.com/7939820/81788575-61c4c900-9520-11ea-834f-422f13b5f8ee.png)

I don't see the value was set to 1 during query time range. Let me know if you need some more logs info.

**Comment by cstyan** [comment]
Without know what time the query execution started from m3db's side I can't say whether there's an issue with this metric or not, but my suspicion is there is not.

Again, that metric only shows the number of remote read queries Prometheus is currently executing. So, if for example your remote read query began and ended execution between two scrape intervals, you would see no value change in that metric. That query you're showing isn't likely to show any change in the value either, since the values you're looking at are 10m apart and your query happened to have ended in between two of the timestamps. it's unlikely that the query execution took more than 6 minutes as it would have had to have to have shown up in the first timestamped 0 value.

If it would be useful, we can discuss adding a `remote_read_queries_total` metric to show total remote read queries executed over time.

**Comment by amitabh91** [comment]
@cstyan Yes I can see a spike for prometheus_remote_storage_remote_read_queries. And as per   @roidelapluie input, it does change from 0 to 1.  

Whenever we query for prometheus_remote_storage_remote_read_queries in Prom UI it will show zero if there are no current read queries is going on.  

We have pointed Grafana to read the metrics from Prometheus which inturn read from M3DB nodes via m3coordinator. We have set storage.tsdb.retention.time is set to 15m in the Prometheus server so that we don't overload Prometheus Pod disk and leverage remote storage. 

We get the metrics data successfully in Grafana on querying for 12 hours, but we got only one spike during those 12 hours.
Attaching a screenshot for reference.
![image](https://user-images.githubusercontent.com/7939820/81912516-4b843f00-95ec-11ea-817c-c3eab3947473.png)

![image](https://user-images.githubusercontent.com/7939820/81903499-76b46180-95df-11ea-9b84-3dd7b036dd17.png)


So I see Prometheus is able to fetch the metric data from remote storage but we don't have any query like remote_read_queries_total which shows total remote read queries executed over time.

**I would suggest we should include a remote_read_queries_total query.**

**Comment by gouthamve** [comment]
The idea is to add a typical latency histogram metric with status codes.

**Comment by cstyan** [comment]
@njingco adding these metrics is probably a good first introduction to the code base for you

**Comment by cstyan** [comment]
fixed in #7328
